sdks/python/apache_beam/ml/inference/tensorrt_inference.py

```diff
@@ -0,0 +1,245 @@
+          'name': name,
+          'dtype': np.dtype(trt.nptype(dtype)),
+          'shape': list(shape),
+          'allocation': allocation,
+          'size': size
+      }
+      self.gpu_allocations.append(allocation)
+      if self.engine.binding_is_input(i):
+        self.inputs.append(binding)
+      else:
+        self.outputs.append(binding)
+    
+    assert self.context
+    assert batch_size > 0
+    assert len(self.inputs) > 0
+    assert len(self.outputs) > 0
+    assert len(self.gpu_allocations) > 0
+
+    for output in self.outputs:
+      self.cpu_allocations.append(np.zeros(output['shape'], output['dtype']))
+
+  def get_engine_attrs(self):
+    """Returns TensorRT engine attributes."""
+    return self.engine, self.context, self.inputs, self.outputs, self.gpu_allocations, self.cpu_allocations
+
+
+class TensorRTEngineHandlerNumPy(ModelHandler[np.ndarray,
+                                              PredictionResult,
+                                              TensorRTEngine]):
+  def __init__(self, min_batch_size: int, max_batch_size: int, **kwargs):
+    """Implementation of the ModelHandler interface for TensorRT.
+
+    Example Usage::
+
+    pcoll | RunInference(
+        TensorRTEngineHandlerNumPy(
+          min_batch_size=1,
+          max_batch_size=1,
+          engine_path="my_uri"))
+
+    Args:
+      min_batch_size: minimum accepted batch size.
+      max_batch_size: maximum accepted batch size.
+      kwargs: Additional arguments like 'engine_path' and 'onnx_path' are
+      currently supported.
+
+    See https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/
+    for details
+    """
+    self.min_batch_size = min_batch_size
+    self.max_batch_size = max_batch_size
+    if 'engine_path' in kwargs:
+      self.engine_path = kwargs.get('engine_path')
+    elif 'onnx_path' in kwargs:
+      self.onnx_path = kwargs.get('onnx_path')
+
+    trt.init_libnvinfer_plugins(TRT_LOGGER, namespace="")
+
+  def batch_elements_kwargs(self):
+    """Sets min_batch_size and max_batch_size of a TensorRT engine."""
+    return {
+        'min_batch_size': self.min_batch_size,
+        'max_batch_size': self.max_batch_size
+    }
+
+  def load_model(self) -> TensorRTEngine:
+    """Loads and initializes a TensorRT engine for processing."""
+    engine = _load_engine(self.engine_path)
+    return TensorRTEngine(engine)
+
+  def load_onnx(self) -> Tuple[trt.INetworkDefinition, trt.Builder]:
+    """Loads and parses an onnx model for processing."""
+    return _load_onnx(self.onnx_path)
+
+  def build_engine(self, network: trt.INetworkDefinition, builder: trt.Builder) -> TensorRTEngine:
+    """Build an engine according to parsed/created network."""
+    engine = _build_engine(network, builder)
+    return TensorRTEngine(engine)
+
+  def run_inference(
+      self,
+      batch: np.ndarray,
+      engine: TensorRTEngine,
+      inference_args: Optional[Dict[str, Any]] = None
+  ) -> Iterable[PredictionResult]:
```

Comment:
I'm seeing this lint error:
```
19:21:43 apache_beam/ml/inference/tensorrt_inference.py:196: error: Argument 1 of "run_inference" is incompatible with supertype "ModelHandler"; supertype defines the argument type as "Sequence[ndarray[Any, Any]]"  [override]
19:21:43 apache_beam/ml/inference/tensorrt_inference.py:196: note: This violates the Liskov substitution principle
19:21:43 apache_beam/ml/inference/tensorrt_inference.py:196: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
```
I think we need to adhere to the [interface](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/ml/inference/base.py#L91) for `run_inference` as found in `base.py`. Could we make it `Sequence[np.ndarray]` like it's done in [sklearn_inference.py](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/ml/inference/sklearn_inference.py#L105)? 

This may require some refactor some of the tests. I think particularly how the input examples are defined

Reply:
Noted